src/ServiceManagement/Services/Commands.Utilities/Common/AzureTools/AzureTool.cs

```diff
@@ -94,13 +94,11 @@ private static string GetSdkVersionRegistryValue()
                     {
                         throw new InvalidOperationException(Resources.AzureToolsNotInstalledMessage);
                     }
-                    version = key.GetSubKeyNames().FirstOrDefault(n => n == AzureSdkVersionInfo.SupportAzureSdkVersionRegKey);
 
-                    if (string.IsNullOrEmpty(version) && key.GetSubKeyNames().Length > 0)
-                    {
-                        throw new InvalidOperationException(string.Format(Resources.AzureSdkVersionNotSupported, AzureSdkVersionInfo.SupportAzureSdkVersionRegKey));
-                    }
-                    else if (string.IsNullOrEmpty(version) && key.GetSubKeyNames().Length == 0)
+                    // select the latest version of SDK installed on the machine.
+                    version = key.GetSubKeyNames().Last();
```

Comment:
You're right. Let me change that to LastOrDefault() so it won't throw when an item is not found.